projects/ngneat/reactive-forms/src/lib/types.ts

```diff
@@ -62,11 +62,60 @@ export type KeyValueControls<T extends Obj> = {
   [K in keyof T]: T[K] extends FormControl<T[K]>
     ? FormControl<T[K]>
     : T[K] extends FormGroup<T[K]>
-    ? FormGroup<T[K]>
-    : T[K] extends FormArray<ArrayType<T[K]>>
-    ? FormArray<ArrayType<T[K]>>
-    : AbstractControl<T[K]>;
+      ? FormGroup<T[K]>
+      : T[K] extends FormArray<ArrayType<T[K]>>
+        ? FormArray<ArrayType<T[K]>>
+        : AbstractControl<T[K]>;
 };
 export type ExtractAbstractControl<T, U> = T extends KeyValueControls<any>
   ? { [K in keyof U]: AbstractControl<U[K]> }
   : T;
+
+/*
+ * Convert a Control type or a value type
+ * Leaving non-control types as is
+ * */
+export type ControlValue<T> = T extends FormControl | FormGroup | FormArray | AbstractControl ? T['value'] : T;
+
+/**
+ * Convert an object of a FormGroup's "value" or "controls" to its "value"
+ * */
+export type ControlsValue<T extends object> = {
+  [K in keyof T]: ControlValue<T[K]>
+};
+
+/**
+ * Converts a value / form control to form control
+ * Converting non-control types to AbstractControl of the type
+ *
+ * The intermediate type is to solve the issue of T being any, thus assignable to all condition and resulting in the "any" type.
+ * */
+type ControlOfValueWithPotentialUnion<T> = T extends AbstractControl ? T : T extends number | string ? FormControl<T> : AbstractControl<T>
+export type ControlOfValue<T> = AbstractControl extends ControlOfValueWithPotentialUnion<T> ? AbstractControl<ControlOfValueWithPotentialUnion<T>['value']> : ControlOfValueWithPotentialUnion<T>;
+
+/**
+ * Convert an object of a FormGroup's "value" or "controls" to "controls".
+ * Converting non-control types to AbstractControl of the type
+ * */
+export type ControlsOfValue<T extends Obj> = {
+  [K in keyof T]: ControlOfValue<T[K]>
+};
+
+/**
+ * Use with FormGroup you want a regular FormControl for each property
+ *
+ * @example
+ * new FormGroup<FlatControls<{
+ *   name: string;
+ *   phone: {
+ *      num: number;
+ *      prefix: number;
+ *   };
+ * }>>({
+ *   name: new FormControl<string>(),
+ *   phone: new FormControl<{num: number, prefix: number}>(),
+ * });
+ * */
+export type FlatControls<T extends Object> = {
```

Comment:
could you add some more tests and examples for it?
for example, add a test for nested form and use `getControl`

Reply:
Add a small test to this, and added this to the documentation.